spec/spec_helper_acceptance.rb

```diff
@@ -64,108 +51,55 @@ def run_device(options = { allow_changes: true })
-          on(host, 'yum install -y vim java')
-          on(host, '/opt/puppetlabs/puppet/bin/gem install pry')
-          # set rich data to true for puppet master and agent
-          on(host, "sed -i '/.main.$/a rich_data = true' /etc/puppetlabs/puppet/puppet.conf")
-          on(host, 'service pe-puppetserver restart', acceptable_exit_codes: [0, 1])
-          on host, puppet('plugin', 'download', '--server', host.to_s)
-        end
-      end
-      hosts.each do |host|
-        device_conf = <<-EOS
-[#{device_hostname}]
+    File.open('spec/fixtures/acceptance-device.conf', 'w') do |file|
+      file.puts <<DEVICE
+[sut]
 type cisco_ios
-url file:///#{default[:puppetpath]}/credentials.yaml
-        EOS
-        create_remote_file(default, File.join(default[:puppetpath], 'device.conf'), device_conf)
-        # not all test devices require an enable_password
-        enable_password = "enable_password: #{device_enable_password}" unless device_enable_password.empty?
-        credentials_yaml = <<-EOS
-host: #{device_ip}
-port: 22
-user: #{device_user}
-password: #{device_password}
-#{enable_password}
-EOS
-        create_remote_file(default, File.join(default[:puppetpath], 'credentials.yaml'), credentials_yaml)
-
-        on(host, "echo #{device_ip} #{device_hostname} >> /etc/hosts")
-
-        # install dependencies on to the server
-        apply_manifest(<<PP
-          $puppetserver_service = $facts['pe_server_version'] ? {
-                                      /./     => 'pe-puppetserver',
-                                      default => 'puppetserver'
-                                  }
-          service { $puppetserver_service: }
-          include "cisco_ios::server", "cisco_ios::proxy"
-PP
-                      )
-        on host, puppet('plugin', 'download', '--server', host.to_s)
-        on host, puppet('device', '-v', '--waitforcert', '0', '--user', 'root', '--server', host.to_s), acceptable_exit_codes: [0, 1]
-
-        # Sign certs
-        puppet_version = on host, puppet('--version')
-
-        if version_is_less(puppet_version.output, '5.99')
-          on host, puppet('cert', 'sign', '--all'), acceptable_exit_codes: [0, 24]
-        else
-          on host, 'puppetserver ca sign --all', acceptable_exit_codes: [0, 24]
-        end
-
-        on host, puppet('plugin', 'download', '--server', host.to_s)
-        on host, puppet('device', '-d', '--user', 'root'), acceptable_exit_codes: [0, 1]
-        # Generate the types
-        on host, 'puppet generate types'
-        # restart the server after installing puppet-resource_api
-        on host, 'systemctl restart  pe-puppetserver.service'
-        # Regenerate the types
-        on host, 'puppet generate types'
-        # set pre-requisites, aaa new-model and enable secret such that we don't get locked out of enable mode
-        enable_password = device_enable_password.empty? ? '' : <<-EOS
-ios_config { "enable password":
```

Comment:
The two `ios_config` configurations is a prerequisite for some of the devices which don't have AAA enabled. So after setting up the credentials it would be good to create a manifest with the `enable password` and `enable aaa`, no longer do we need to check if `device_enable_password` is empty as we determine it is safer to set the enable password otherwise we run the risk of locking the devices